### PR TITLE
Convolutional Code 27

### PR DESCRIPTION
**Pull request by ArashYzd**

### CODE REVIEW

File: lib/error_control/error_correction.cpp [hunk 1]
```diff
@@ -254,12 +254,12 @@ namespace ex2 {
       switch(scheme) {
         // CCSDS convolutional codiing is supported.
         case ErrorCorrectionScheme::CCSDS_CONVOLUTIONAL_CODING_R_1_2:
+          isValid = true;
+          break;
         case ErrorCorrectionScheme::CCSDS_CONVOLUTIONAL_CODING_R_2_3:
         case ErrorCorrectionScheme::CCSDS_CONVOLUTIONAL_CODING_R_3_4:
         case ErrorCorrectionScheme::CCSDS_CONVOLUTIONAL_CODING_R_5_6:
         case ErrorCorrectionScheme::CCSDS_CONVOLUTIONAL_CODING_R_7_8:
```
Comment (knud) [inline lib/error_control/error_correction.cpp]: As we discussed, you can put this back.

File: lib/error_control/convCode27.cpp [hunk 2]
```diff
@@ -79,7 +82,16 @@ namespace ex2 {
       /* Decode block */
       uint8_t encodedArr[8*framebits+constraint_length-1];
       for (int i; i<sizeof(encodedArr); i++){
-        encodedArr[i] = offset + amp*((encodedPayload[i/8] >> (i%8)) & 1);//The viterbi decoder makes a decision based on the 127 threshold.
+        #if QA_NOISY_TEST
```
Comment (knud) [inline lib/error_control/convCode27.cpp:85]: I can see why we'd do the noisy test here, but I'd prefer to do it in the unit test. 

It brings up an interesting question of quantization. On the UHF radio, we get bits, so the noise is very much quantized. On the ground, if we use a B200mini or other SDR, we get float samples, and could to "softer" decoding. We better discuss.

File: lib/error_control/convCode27.cpp [hunk 3]
```diff
@@ -59,8 +62,8 @@ namespace ex2 {
     convCode27::decode(const PPDU_u8::payload_t& encodedPayload, float snrEstimate,
```
Comment (knud) [inline lib/error_control/convCode27.cpp:62]: There is a fundamental assumption here that needs discussing. Along the lines of my comment below, we get from the UHF radio data that a hard decision has been applied to, in effect. The encodedPayload is not 1 symbol per byte, it's packed, 8 symbols per byte. We need to discuss

File: lib/error_control/sim.c [hunk 4]
```diff
@@ -0,0 +1,46 @@
+#include <math.h>
```
Comment (knud) [inline lib/error_control/sim.c]: This should probably be in the `math` directory.

**Review by knud — APPROVED** [review]

File: lib/error_control/convCode27.cpp (hunk 2 above)
Comment (ArashYzd) [reply lib/error_control/convCode27.cpp:85]: Yes, unfortunately, I couldn't find a way to get around this. The decoder is implemented in a way to make the best decision based on the sampled signals and not quantified bits. So, unless the input argument of the decode function is those samples (no need to convert encodedPayload to encodedArr like below), it's not a simple way to put this test in the unit tests.

File: lib/error_control/convCode27.cpp (hunk 3 above)
Comment (ArashYzd) [reply lib/error_control/convCode27.cpp:62]: I'm aware, and the best solution I found was to turn them into 8byte samples with a low confidence level (close to threshold) to minimize the effect of noise. My schedule is open if you'd like to discuss this.

**Pull request merged**